boto/s3/checksums.py

```diff
@@ -107,4 +108,12 @@ def set_checksum_header(checksum, provider, headers,
             headers[provider.checksum_sha1_header] = calculated_checksum
         elif checksum_lower == 'sha256' and provider.checksum_sha256_header not in headers:
             headers[provider.checksum_sha256_header] = calculated_checksum
+        if config.getbool('Boto', 'either_md5_checksum_headers', True):
+            if len(headers.keys() & {provider.checksum_crc32_header,
+                                     provider.checksum_crc32c_header,
+                                     provider.checksum_crc64nvme_header,
+                                     provider.checksum_sha1_header,
+                                     provider.checksum_sha256_header}) > 0:
```

Comment:
I find this code a little too confusing and hard to read. Please consider the following:
```
checksumHeaders = {provider.checksum_crc32_header, ...}
if len(set(headers.keys()).intersection(checksumHeaders)) > 0):
    ...
```